src/runtime/vdom/vdom-render.ts

```diff
@@ -723,8 +726,17 @@ const markSlotContentForRelocation = (elm: d.RenderNode) => {
 
         // check that the node is not a content reference node or a node
         // reference and then check that the host name does not match that of
-        // childNode
-        if (!node['s-cn'] && !node['s-nr'] && node['s-hn'] !== childNode['s-hn']) {
+        // childNode.
+        // In addition, check that the slot either has not already been relocated, or
+        // that its current location's host is not childNode's host. This is essentially
+        // a check so that we don't try to relocate (and then hide) a node that is already
+        // where it should be.
+        if (
+          !node['s-cn'] &&
+          !node['s-nr'] &&
+          node['s-hn'] !== childNode['s-hn'] &&
+          (!BUILD.experimentalSlotFixes || !node['s-sh'] || node['s-sh'] !== childNode['s-hn'])
```

Comment:
@alicewriteswrongs  Because we only want the rest of that check to happen if `experimentalSlotFixes` is enabled. So, if it isn't, we'll short-circuit and enter the code block before evaluating the new `s-sh` element property.

Reply:
ohhh ok that makes sense - so the code block is always relevant (slot fixes or not) but that `s-sh` business is gated behind `experimentalSlotFixes` - makes sense!